libcontainer/system/rlimit_linux_go122.go

```diff
@@ -0,0 +1,25 @@
+//go:build !go1.23
+
+package system
+
+import (
+	"sync/atomic"
+	"syscall"
+	_ "unsafe" // Needed for go:linkname to work.
+)
+
+//go:linkname syscallOrigRlimitNofile syscall.origRlimitNofile
+var syscallOrigRlimitNofile atomic.Pointer[syscall.Rlimit]
+
+// ClearRlimitNofileCache clears go runtime's nofile rlimit cache.
```

Comment:
When I was choosing go linkname hack, I really want to add a comment like this:
FIXME: Once golang has a public API to clear the nofile rlimit cache, we should remove this hack and switch to the official API.
But it seems like there is no chance to add this specific API in golang, so it's no need to add this similar comment, but maybe we can add a comment to inform other contributors that we should not work around for the `hall of shame` anymore. For example:
// Note: we should remove this hack once runc doesn't support 1.22 anymore.

Reply:
I agree, but the facts that

 - the file name contains `_go122`;
 - the file has `//go:build !go1.23`

makes it obvious that this is to be removed once go 1.22 is no longer supported. Having a comment saying so would be redundant.

Any way, redundancy is not always bad. Added.